common/txmgr/types/priority_queue.go

```diff
@@ -0,0 +1,115 @@
+package types
+
+import (
+	"sync"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+type PriorityQueue[
```

Comment:
This type seems internal to the TXM right?
I mean not exposed to anywhere else outside the TXM.
Could you not export it (rename to priorityQueue) and move it out of types package?
The types package I think is for types we want to export to other packages.